frame/transaction-payment/src/lib.rs

```diff
@@ -234,7 +259,8 @@ impl<T: Trait + Send + Sync> SignedExtension for ChargeTransactionPayment<T>
 				Ok(imbalance) => imbalance,
 				Err(_) => return InvalidTransaction::Payment.into(),
 			};
-			T::OnTransactionPayment::on_unbalanced(imbalance);
+			let (tip_imbalance, fee_imbalance) = imbalance.split(tip);
+			T::OnTransactionPayment::on_unbalanceds(PairIterator::with_two(fee_imbalance, tip_imbalance));
```

Comment:
I think we should add to the doc of `type OnTransactionPayment` that unbalance will be iterator of fee and then tip